File: euler/problem002.go
package euler

func fibImp(limit int) int {
	a := 1
	b := 1
	s := 0
	for b <= limit {
		if b%2 == 0 {
			s += b
		}
		a, b = b, a+b
	}
	return s
}

func fibOpt(limit int) int {
	a := 1
	b := 1
	c := 2
	s := 0
	for c <= limit {
		s += c
		a = b + c
		b = a + c
		c = a + b
	}
	return s
}

File: julia/Problem002.jl
function fibIter(limit)
  a = 0
  b = 1
  acc = 0
  while (a + b) <= limit
    next = a + b
    if iseven(next)
      acc += next
    end
    a, b = b, a+b
  end
  acc
end

function fibOpt(limit)
  a = 1
  b = 1
  c = 2
  acc = 0
  while c <= limit
    acc += c
    a = b + c
    b = a + c
    c = a + b
  end
  acc
end
